### PR TITLE
feat!: `onDeploy` fuels config supports all Sway program types

### LINKED ISSUE #3356: `onDeploy` does not work for predicates and scripts in `fuels deploy`
It would be interesting to implement this for predicates and scripts to enable some automations. Currently,`onDeploy` is executed after contracts are deployed.
https://github.com/FuelLabs/fuels-ts/blob/88d7535733c98340cd21bc6f7846290c462bef4c/packages/fuels/src/cli/commands/deploy/index.ts#L11-L18

My suggestion would be to implement something like this:

```ts
export async function deploy(config: FuelsConfig) {
  /**
   * Deploy contract and save their IDs to JSON file.
   */
  const contractIds = await deployContracts(config);
  await saveContractIds(contractIds, config.output);

  /**
   * Deploy scripts and save deployed files to disk.
   */
  const scripts = await deployScripts(config);
  saveScriptFiles(scripts, config);

  /**
   * Deploy predicates and save deployed files to disk.
   */
  const predicates = await deployPredicates(config);
  savePredicateFiles(predicates, config);
  
  config.onDeploy?.(config, {
    scripts,
    predicates,
    contracts: contractIds,
  });

  /**
   * After deploying scripts/predicates, we need to
   * re-generate factory classe with the loader coee
   */
  await generateTypes(config);

  return contractIds;
}
```

### PR DESCRIPTION
**Pull request by YaTut1901**
- Closes #3356

# Breaking Changes

- Changed the outputted data from the `onDeploy` callback method for the `fuels.config.ts`. Instead of just emitting the deployed contracts (as an array), it will now emit an object with `contracts`, `predicates` and `scripts`.

```ts
// Before (fuels.config.ts)
import { createConfig, FuelsConfig, DeployedContract } from 'fuels';

export default createConfig({
  output: 'dir/out',
  onDeploy: (config: FuelsConfig, deployedContracts: DeployedContract[]) => {
    console.log('contracts', deployedContracts);
  }
});
```

```ts
// After (fuels.config.ts)
import { createConfig, FuelsConfig, DeployedData } from 'fuels';

export default createConfig({
  output: 'dir/out',
  onDeploy: (config: FuelsConfig, deployed: DeployedData[]) => {
    console.log('contracts', deployed.contracts);
    console.log('predicates', deployed.predicates);
    console.log('scripts', deployed.scripts);
  }
});
```

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: .changeset/khaki-trees-wink.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"fuels": minor
+---
+
+onDeploy callback updated to support automation with scripts and predicates additionally to contracts
```
Comment (petertonysmith94) [inline .changeset/khaki-trees-wink.md:5]:
```suggestion
feat!: `onDeploy` callback now supports contracts, scripts and predicates
```

You'll want to update the title to match :)

**Review by petertonysmith94 — COMMENTED** [review]
@YaTut1901 lovely work 🚀 

You may want to update the description to include a section on breaking changes.

This forms part of our [migration guides](https://docs.fuel.network/docs/migrations-and-disclosures/migrations/typescript-sdk/) in the docs, we generally have a `before` and `after` section to show the migration steps :)

File: .changeset/khaki-trees-wink.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"fuels": minor
+---
+
+feat!: `onDeploy` callback now supports contracts, scripts and predicates
```
Comment (petertonysmith94) [inline .changeset/khaki-trees-wink.md:5]:
```suggestion
feat!: `onDeploy` fuels config supports all Sway program types
```

Apologies, too many characters

**Review by petertonysmith94 — APPROVED** [review]
Nice job @YaTut1901 - thanks for the contribution 🙏🏼

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**